src/server.rs

```diff
@@ -241,13 +244,41 @@ struct Cli {
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let contents = fs::read_to_string("./config.toml")
-        .expect("Something went wrong reading the file");
-
-    let config: Config = toml::from_str(&contents)
-        .expect("Could not deserialize the config file");
-
-    println!("{:#?}", config);
+    // Attempt to read the contents of the config file
+    let contents = fs::read_to_string("../../config.toml");
+    match contents {
+        Ok(contents) => {
+            // Open a log file for writing
+            let mut log_file = File::create("../../config_parsing.log")?;
+
+            // Attempt to deserialize the config file
+            let config: Result<Config, _> = toml::from_str(&contents);
+            match config {
+                Ok(config) => {
+                    println!("{:#?}", config);
+                    // Write the parsed configuration to the log file
+                    writeln!(log_file, "{:#?}", config)?;
+                },
+                Err(err) => {
+                    // Log the error to the file
+                    writeln!(log_file, "Could not deserialize the config file: {:?}", err)?;
+                    // Print the error to stderr
+                    eprintln!("Could not deserialize the config file: {:?}", err);
+                    // Handle the error 
+                    return Err(err.into());
+                }
+            }
+        },
+        Err(err) => {
+            // Log the error to the file
+            let mut log_file = File::create("../../config_parsing.log")?;
+            writeln!(log_file, "Something went wrong reading the file: {:?}", err)?;
+            //Print the error to stderr
+            eprintln!("Something went wrong reading the file: {:?}", err);
```

Comment:
Yeah same we can rely on the logging file as soon as it’s created unless we have `File::create` itself returning an error just above L274 ?

Reply:
Yeah, im fairly certain the file isnt created it will still print to stdout.